### PR TITLE
PUBDEV-3201: Adds support to export to multiple part files to Python API

### PR DESCRIPTION
**Pull request by michalkurka**
The pull request brings the Python API up to date with the R API, it completes jira PUBDEV-3201.

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/120)

### CODE REVIEW

File: h2o-py/h2o/h2o.py [hunk 1]
```diff
@@ -849,19 +849,21 @@ def load_model(path):
...
     :param path: the path to the save point on disk.
     :param force: if True, overwrite any preexisting file with the same path
+    :param parts: enables export to multiple 'part' files instead of just a single file.
+        Convenient for large datasets that take too long to store in a single file.
+        Use parts=-1 to instruct H2O to determine the optimal number of part files or
+        specify your desired maximum number of part files. Path needs to be a directory
+        when exporting to multiple files.
+        Default is to export to a single file (parts=1).
     """
-    assert_is_type(frame, H2OFrame)
```
Comment (st-pasha) [inline h2o-py/h2o/h2o.py]: Please don't remove these typechecks -- they guarantee that the user does not abuse the calling conventions.

Comment (michalkurka) [inline h2o-py/h2o/h2o.py]: Good catch! This was removed in an invalid merge conflict resolution. Thank you

File: h2o-py/tests/testdir_misc/pyunit_export_file_multi.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    pros_hex[5] = pros_hex[5].asfactor()
+    pros_hex[8] = pros_hex[8].asfactor()
+
+    p_sid = pros_hex.runif()
+    pros_train = pros_hex[p_sid > .2, :]
+    pros_test = pros_hex[p_sid <= .2, :]
+
+    glm = H2OGeneralizedLinearEstimator(family="binomial")
+    myglm = glm.train(x=list(range(2, pros_hex.ncol)), y=1, training_frame=pros_train)
+    mypred = glm.predict(pros_test)
```
Comment (st-pasha) [inline h2o-py/tests/testdir_misc/pyunit_export_file_multi.py]: Not sure why you need to spend time training a GLM model, if all you test is frame exporting. Can't you just export the original pros_hex? Or better yet, create a random frame using `h2o.create_frame` of desired dimension and export that? Ideally we want each test to be as small as possible...

File: h2o-py/tests/testdir_misc/pyunit_export_file_multi.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    print(py_pred.head())
+    h_pred = mypred.as_data_frame(True)
+    print(h_pred.head())
+
+    #Test to check if py_pred & h_pred are identical
+    try:
+        assert_frame_equal(py_pred,h_pred)
+        return True
+    except:
+        return False
```
Comment (st-pasha) [inline h2o-py/tests/testdir_misc/pyunit_export_file_multi.py]: Hmm, you catching the AssertionError here, and turn it into a `False` return value. But as far as I know return values from test functions are ignored. Thus, this check effectively becomes a noop

File: h2o-py/h2o/h2o.py [hunk 4]
```diff
@@ -849,19 +849,21 @@ def load_model(path):
...
-    assert_is_type(path, str)
-    assert_is_type(force, bool)
-    H2OJob(api("GET /3/Frames/%s/export/%s/overwrite/%s" % (frame.frame_id, path, str(force).lower())),
+    H2OJob(api("POST /3/Frames/%s/export" % (frame.frame_id), data={"path": path, "num_parts": parts, "force": force}),
```
Comment (st-pasha) [inline h2o-py/h2o/h2o.py]: Nice that you got rid of that horrible "GET /3/Frames/%s/export/%s/overwrite/%s" endpoint (who knows what happens when the user provides a path containing special characters, spaces, dots, etc)

File: h2o-py/tests/testdir_misc/pyunit_export_file_multi.py (hunk 2 above)
Comment (michalkurka) [inline h2o-py/tests/testdir_misc/pyunit_export_file_multi.py]: Well, I agree - this doesn't make much sense to me either. This test is not actually new - it is a modified version of an existing test. I think I would prefer to keep it this way - to mimmic the original test and also the same test we have in R.

File: h2o-py/tests/testdir_misc/pyunit_export_file_multi.py (hunk 3 above)
Comment (michalkurka) [inline h2o-py/tests/testdir_misc/pyunit_export_file_multi.py]: Fixed here and in the original test. Thanks

**Comment by st-pasha** [comment]
ok looks good

**Pull request merged**